temporalio/worker/replayer.py

```diff
@@ -122,28 +105,127 @@ async def replay_workflow(
                 max_task_queue_activities_per_second=None,
             ),
         )
-        workflow_worker = _WorkflowWorker(
-            bridge_worker=lambda: bridge_worker,
+        self._workflow_worker = _WorkflowWorker(
+            bridge_worker=lambda: self._bridge_worker,
             namespace=self._config["namespace"],
-            task_queue=started_event.workflow_execution_started_event_attributes.task_queue.name,
+            task_queue=tq,
             workflows=self._config["workflows"],
             workflow_task_executor=self._config["workflow_task_executor"],
             workflow_runner=self._config["workflow_runner"],
             data_converter=self._config["data_converter"],
             interceptors=self._config["interceptors"],
             debug_mode=self._config["debug_mode"],
-            fail_on_eviction=True,
+            on_eviction_hook=self._replayer_eviction_hook(
+                fail_fast=self._config["fail_fast"]
+            ),
         )
+        self._current_run_results = WorkflowReplayResults(False, dict())
+
+    def config(self) -> ReplayerConfig:
+        """Config, as a dictionary, used to create this replayer.
+
+        Returns:
+            Configuration, shallow-copied.
+        """
+        config = self._config.copy()
+        config["workflows"] = list(config["workflows"])
+        return config
 
-        # Run it
-        try:
-            await workflow_worker.run()
-        finally:
-            # We must finalize shutdown here
+    async def replay_workflow(
+        self,
+        history: WorkflowHistory,
+    ) -> None:
+        """Replay a workflow for the given history.
+
+        Args:
+            history: The history to replay. Can be fetched directly, or use
+              :py:meth:`WorkflowHistory.from_json` to parse a history downloaded via `tctl` or the
+              web ui.
+        """
+
+        async def gen_hist():
+            yield history
+
+        await self.replay_workflows(gen_hist())
+
+    async def replay_workflows(
+        self, histories: AsyncIterator[WorkflowHistory]
+    ) -> WorkflowReplayResults:
```

Comment:
Not sure I agree. I'm just having a hard time imagining what the use case is there, and getting a summary is going to be easier for most people most of the time. If they want to fail fast, we have the option for that. If they use the iterator though, they need to do some kind of `results.any(|i| i.failure.exists())` deal to check if there was any failure, which isn't hard per se, but a lot less convenient than just `results.had_failure`

The returned thing would also need run id, if it became an iterator.

Reply:
Streaming workflows to histories to replays is a reasonable method of operation I think. I also wouldn't be surprised if one day we will start giving back more on replay instead of just errors, e.g. the return type, ability to execute query, etc.

If you are concerned about an async iterator being too hard for users to use (despite the fact that we accept one), you could make `WorkflowReplayResults` be an async iterator itself (ref https://peps.python.org/pep-0492/#asynchronous-iterators-and-async-for) and have a `async def collect_results()` that is a shortcut.

If you don't want to mess w/ all of this now I can do it a bit later when I move history and do async iterators for listing and such.

> The returned thing would also need run id, if it became an iterator.

Do we return in order 1-by-1? I think that's good enough, but I think a run ID, and really the whole workflow history would have value no such a result. So could have `WorkflowReplayResult` have `failure: Optional[Exception]` and `history: WorkflowHistory` and add a getter `@property def run_id(self) -> str` on `WorkflowHistory` that extracted the run ID from the event sequence.

Reply:
I will come back and do this later